spec_validator.go

```diff
@@ -0,0 +1,148 @@
+/**
+ *  Copyright (c) 2022  Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wrp
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/google/uuid"
+)
+
+const (
+	serialPrefix = "serial"
+	uuidPrefix   = "uuid"
+	eventPrefix  = "event"
+	dnsPrefix    = "dns"
+)
+
+var (
+	ErrorInvalidMessageEncoding = errors.New("invalid message encoding")
+	ErrorInvalidMessageType     = errors.New("invalid message type")
+	ErrorInvalidSource          = errors.New("invalid Source name")
+	ErrorInvalidDestination     = errors.New("invalid Destination name")
+)
+
+// SpecValidators is a WRP validator that ensures messages are valid based on
+// each spec validator in the list. Only validates the opinionated portions of the spec.
+// SpecValidators validates the following fields: UTF8 (all string fields), MessageType, Source, Destination
+func SpecValidators() Validators {
+	return Validators{UTF8Validator, MessageTypeValidator, SourceValidator, DestinationValidator}
+}
+
+// UTF8Validator is a WRP validator that takes messages and validates that it contains UTF-8 strings.
+var UTF8Validator ValidatorFunc = func(m Message) error {
```

Comment:
Please make all of these `var ValidatorFunc`s into `func`s like `SpecValidators()` so they can't be modified by consumers.  Then move `locatorPattern` to the top so all package level vars are listed before any functions.  Then the general file structure will match the rest of our codebase.